src/DynamoCore/Models/HomeWorkspaceModel.cs

```diff
@@ -20,6 +20,13 @@ public class HomeWorkspaceModel : WorkspaceModel
 
         private bool graphExecuted;              
         public readonly bool VerboseLogging;
+       
+        /// <summary>
+        /// This is set from Dynamoviewmodel. This is required here otherwise 
+        /// preview graph will run under every context that affects some of the test.
+        /// Also preview graph should run only when showrunpreview is set to true.
+        /// </summary>        
+        public bool ShowRunPreview { get; set; }
```

Comment:
Yes. Model should not know about ViewModel. But preview graph should run only when ShowRunPreview is checked on the settings menu. In the model layer this is always false, hence PreviewGraph would never run. The other way is set the ShowRunPreview on DynamoModel and WorkspaceModel can get the value from DynamModel. I thought this is straight forward. Please let me know if you have any other thoughts around this.